### ISSUE TITLE
Suggestion: Reduce size of rendered HTML-elements

### ISSUE THREAD

**Issue by EMaderbacher** [chanan/BlazorStrap#73]
To reduce amount of data sent to Brower in created HTML-Elements are a lot of attributes which shouldn't be rendered when they've default values.
For example 
```html
<input class="  form-control  valid" type="text" size="0" selectedIndex="0" value="">
```
contains several attributes which aren't needed / shouldn't be rendered.
Blazor itself is also not rendering such attributes, see https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.0#razor-support at topic **Conditional HTML element attributes**.

Additionally the class attribute contains unnecessary spaces which could be removed from created html.
Just tell me if such "optimizations" are ok for you.

EDITED: I fixed the url to the docs, for some reason it was not linked right in the markdown

**Comment by chanan** [comment]
Yes, these are good suggestions.

For the class name, I am considering redoing how they are generated and using a C# class to generate them (and at the same time clean up the output). Either directly using: https://github.com/EdCharbeneau/CssBuilder or similar to it.

Is that something you want to contribute?

**Comment by EMaderbacher** [comment]
Hi chanan,

for the classname I was thinking about a simpler solution.
For getting the classname you've for example in BlazorInput currently:
```c#
return $"{isvalid} {isinvalid} {cls} {size} {Class}";
```
I suggest here just a minor modification:
```c#
return $"{isvalid} {isinvalid} {cls} {size} {Class}".RemoveUnneededSpaces();
```
The extension method ```RemoveUnneededSpaces``` will remove all unneeded Space-Characters from return value.
How do you think about that?

Attributes with default values:
The fastest solution is to change the parameters to nullable properties, for example:
```c#
[Parameter] private int? SelectSize { get; set; }
```
Is this something what is acceptable by you?

I can contribute this two changes. Just give feedback if both or only one of my suggestions is ok

**Comment by chanan** [comment]
Go for it!

**Comment by chanan** [comment]
Done by @EMaderbacher in PR: #78 

Released in 1.0.0-preview6-14
